### PR TITLE
Update trunk to Starboard 17

### PR DESCRIPTION
**Pull request by madhurajayaraman**
b/355449840

### CODE REVIEW

**Review by kaidokert — COMMENTED** [review]
This LGTM - but i thought we also need to add new matching `sabi` json files with this ?

File: starboard/sabi/arm/hardfp/sabi-v17.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "variables": {
+    "sb_api_version": 16,
```
Comment (andrewsavage1) [inline starboard/sabi/arm/hardfp/sabi-v17.json:3]: This needs to be incremented (as in the other sabi files)

File: starboard/nplb/nplb_evergreen_compat_tests/sabi_test.cc [hunk 2]
```diff
@@ -197,6 +197,55 @@ const char* kSabiJsonIdX64Sysv =
     "\"target_arch\":\"x64\",\"target_arch_sub\":\"\",\"word_size\":64}";
 #endif  // SB_API_VERSION == 16
 
+#if SB_API_VERSION == 17
+const char* kSabiJsonIdArmHardfp =
+    "{\"alignment_char\":1,\"alignment_double\":8,\"alignment_float\":4,"
+    "\"alignment_int\":4,\"alignment_llong\":8,\"alignment_long\":4,"
+    "\"alignment_pointer\":4,\"alignment_short\":2,\"calling_convention\":"
+    "\"eabi\",\"endianness\":\"little\",\"floating_point_abi\":\"hard\","
+    "\"floating_point_fpu\":\"vfpv3\",\"sb_api_version\":16,\"signedness_of_"
```
Comment (andrewsavage1) [inline starboard/nplb/nplb_evergreen_compat_tests/sabi_test.cc:206]: You'll have to update the sb_api_version here as well

Comment (madhurajayaraman) [reply starboard/nplb/nplb_evergreen_compat_tests/sabi_test.cc:206]: I did, it grabs the value from https://github.com/youtube/cobalt/pull/3884/files#diff-989138430bc663c7b2bb63806b3c9252735f567838cf7a6438626ea1aa1ea4b3R9
Missed it in the first set of changes, so it's rebuilding rn.

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**